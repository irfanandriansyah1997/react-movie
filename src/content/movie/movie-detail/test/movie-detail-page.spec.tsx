import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';

import MovieDetailPage from '../movie-detail.page';
import StoreBuilder from '../../../../shared/modules/redux/builder/store.builder';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
    useParams: (): Record<string, string> => ({
        id: 'tt1569923',
        teamId: 'team-id1',
    })
}));

describe('Testing Snapshot Pages', () => {
    it('should render card movie and search bar', (done: () => void) => {
        const store = StoreBuilder.singleton();

        const pages = mount(
            <HashRouter>
                <Provider store={store}>
                    <MovieDetailPage />
                </Provider>
            </HashRouter>
        );

        setTimeout(() => {
            expect(pages.html()).toMatchSnapshot();
            done();
        }, 3000);
    });
});