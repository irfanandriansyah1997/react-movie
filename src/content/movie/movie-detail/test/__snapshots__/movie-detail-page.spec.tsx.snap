// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Snapshot Pages should render card movie and search bar 1`] = `"<div class=\\"ui-container\\"><div class=\\"detail\\"><div class=\\"heading\\"><div class=\\"ui-container\\"><a href=\\"#/\\"><span class=\\"material-icons\\">arrow_back_ios</span><p class=\\"text heading\\" style=\\"font-weight: 400;\\">Kembali</p></a></div></div><img src=\\"https://m.media-amazon.com/images/M/MV5BNmY4ZDZjY2UtOWFiYy00MjhjLThmMjctOTQ2NjYxZGRjYmNlL2ltYWdlL2ltYWdlXkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg\\" alt=\\"Batman: Under the Red Hood\\"><table><tbody><tr><td>Title</td><td>Batman: Under the Red Hood</td></tr><tr><td>Genre</td><td>Animation, Action, Crime, Drama, Sci-Fi, Thriller</td></tr><tr><td>IMDB ID</td><td>tt1569923</td></tr><tr><td>Figure</td><td><ul class=\\"actors\\"><li><div>Actors</div><div>Bruce Greenwood, Jensen Ackles, John DiMaggio, Neil Patrick Harris</div></li><li><div>Directors</div><div>Brandon Vietti</div></li><li><div>Writters</div><div>Judd Winick, Bob Kane (Batman created by)</div></li></ul></td></tr><tr><td>Plot</td><td>There's a mystery afoot in Gotham City, and Batman must go toe-to-toe with a mysterious vigilante, who goes by the name of Red Hood. Subsequently, old wounds reopen and old, once buried memories come into the light.</td></tr><tr><td>Production</td><td>WARNER BROTHERS PICTURES</td></tr><tr><td>Released</td><td>27 Jul 2010</td></tr><tr><td>Runtime</td><td>75 min</td></tr><tr><td>Type</td><td>movie</td></tr><tr><td>Rated</td><td>PG-13</td></tr><tr><td>Rating</td><td><ul><li><div>Internet Movie Database</div><div>8.1/10</div></li><li><div>Rotten Tomatoes</div><div>100%</div></li></ul></td></tr></tbody></table></div></div>"`;
